src/applications/static-pages/facilities/FacilityMapWidget.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import FacilityApiAlert from './FacilityApiAlert';
+import { mapboxToken } from '../../facility-locator/utils/mapboxToken';
+import { buildAddressArray } from '../../facility-locator/utils/facilityAddress';
+import environment from '../../../platform/utilities/environment';
+import { connect } from 'react-redux';
+
+export class FacilityMapWidget extends React.Component {
+  render() {
+    if (this.props.loading) {
+      return <LoadingIndicator message="Loading facility..." />;
+    }
+
+    if (this.props.error) {
+      return <FacilityApiAlert />;
+    }
+
+    const facilityDetail = this.props.facility;
+
+    const lat = facilityDetail.attributes.lat;
+    const long = facilityDetail.attributes.long;
+    let address = buildAddressArray(facilityDetail);
+
+    if (address.length !== 0) {
+      address = address.join(', ');
+    } else {
+      // If we don't have an address fallback on coords
+      address = `${lat},${long}`;
+    }
+
+    const pinURL = encodeURIComponent(
+      `${environment.BASE_URL}/img/icons/health-pin.png`,
+    );
+
+    const mapUrl = `https://api.mapbox.com/v4/mapbox.streets/url-${pinURL}(${long},${lat})/${long},${lat},16/500x300.png?access_token=${mapboxToken}`;
+    const mapLink = `https://maps.google.com?saddr=Current+Location&daddr=${address}`;
+    const imageLink = document.getElementById('google-map-link-image');
+
+    if (imageLink) {
+      imageLink.setAttribute('href', mapLink);
```

Comment:
Generally, React recommends against manipulating the DOM or doing any other side effects in the `render` [method](https://reactjs.org/docs/react-component.html#render).

The alternative is to put this logic in a [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate) lifecycle method:

```jsx
// Runs every time the render method would run
componentDidUpdate(prevProps) {
  // We only want to run this logic when we were loading before, but now we're done loading and there's no error
  if (prevProps.loading && !this.props.loading && !this.props.error) {
     const facilityDetail = this.props.facility;
     const lat = facilityDetail.attributes.lat;
     const long = facilityDetail.attributes.long;

     let address = buildAddressArray(facilityDetail);
     if (address.length !== 0) {
      address = address.join(', ');
    } else {
      // If we don't have an address fallback on coords
      address = `${lat},${long}`;
    }
    const imageLink = document.getElementById('google-map-link-image');
    if (imageLink) {
      imageLink.setAttribute('href', mapLink);
    }
  }
}
```